### PR TITLE
Issues 84

### PR DESCRIPTION
**Pull request by samarthdd**
Completed Issue -https://github.com/k8-proxy/k8-test-data/issues/84

Fetching files from minio

Zipping to .zip, .7z , .rar , .gz , .tar 

Storing in s3 bucket

Updated release notes

### CODE REVIEW

File: compressed_file_generator/main.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        try:
+            count = 0
+            bucket_list = self.minio_client.list_buckets()
+            for bucket in bucket_list:
+                logger.info(f'bucker_name : {bucket} ')
+                objects = self.minio_client.list_objects(bucket)
+                for object in objects:
+
+                    count = count + 1
+                    if not object.split(".")[-1]=="zip":
```
Comment (susilupwork) [inline compressed_file_generator/main.py:31]: Get only the Glasswall approved file type as mentioned by Adam Sloman .  https://glasswallsolutions.com/technology/

File: compressed_file_generator/main.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        try:
+            logger.info(f'zipping file {file_path} to .rar compression')
+            rar_name = Compress.rar_compresstion(target, file_name, file_path)
+            bucket_name = 'rar'
+            target_file_path = target + file_name + '.' + bucket_name
+            self.s3_client.upload_file(file_path=target_file_path, file_name=rar_name, bucket=bucket_name)
+        except Exception as err:
+            logger.error(f'Process:compress_and_upload: .rar error {err}')
+            raise Exception(f'Error while .rar compression and uploading {err}')
+
```
Comment (susilupwork) [inline compressed_file_generator/main.py:101]: Coding standard violation, too lengthy method. Break in to two or more functions.

File: compressed_file_generator/src/utils/minio_service.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import logging
```
Comment (susilupwork) [inline compressed_file_generator/src/utils/minio_service.py:1]: This is the 4th Minio Service client side code.  Rather create a util package where keep all client code. Call them in your respective modules.

File: compressed_file_generator/.env_sample [hunk 4]
```diff
@@ -0,0 +1,10 @@
+MINIO_URL=
+MINIO_ACCESS_KEY=
+MINIO_SECRET_KEY=
+S3_ENDPOINT=
+S3_ACCESS_KEY_ID=
+S3_SECRET_ACCESS_KEY=
+S3_REGION=
+PYTHONUNBUFFERED=
+S3_BUCKET=
+NUM_OF_FILES=100
```
Comment (susilupwork) [inline compressed_file_generator/.env_sample:10]: Use .env file only for keys.

**Review by susilupwork — CHANGES_REQUESTED** [review]
- Get only the Glasswall approved file type as mentioned by Adam Sloman . https://glasswallsolutions.com/technology/
- 
- This is the 4th Minio Service client side code.  Rather create a util package where keep all client code. Call them in your respective modules. Same goes for S3 client python file.

File: compressed_file_generator/main.py (hunk 1 above)
Comment (samarthdd) [reply compressed_file_generator/main.py]: For time being keeping all formats as per Adam Sloman suggestion .

File: compressed_file_generator/main.py (hunk 2 above)
Comment (samarthdd) [reply compressed_file_generator/main.py:52]: I have changed this part and made it as 2 function call

File: compressed_file_generator/.env_sample (hunk 4 above)
Comment (samarthdd) [reply compressed_file_generator/.env_sample]: Removed PYTHONUNBUFFERED=

NUM_OF_FILES should come from user.So i have kept it in env

File: compressed_file_generator/src/utils/minio_service.py (hunk 3 above)
Comment (samarthdd) [reply compressed_file_generator/src/utils/minio_service.py:1]: All Minio Service uses minio library and customise it according to project need

These are all custom code when I investigated all. In this particular case, i need to check types before proceeding furthur which is not the case in others

I have removed extra codes and keeping what is required

File: compressed_file_generator/src/utils/s3_client.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import logging
```
Comment (susilupwork) [inline compressed_file_generator/src/utils/s3_client.py:1]: Do not copy paste s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.  Or you need extra method update in library.  Same goes for Minio also.

**Review by susilupwork — CHANGES_REQUESTED** [review]
Do not copy paste s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.  Or you need extra method update in library.  Same goes for Minio also.

File: compressed_file_generator/main.py (hunk 1 above)
Comment (samarthdd) [reply compressed_file_generator/main.py]: I have removed minio and s3  service code and calling microservice

**Review by susilupwork — APPROVED** [review]
Looks good

**Review by dtollaku — APPROVED** [review]

**Pull request merged**